// ng2-nvd3/compiled/lib/ng2-nvd3.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '../../lib/ng2-nvd3.component';
const styles_NvD3Component:any[] = ([] as any[]);
export const RenderType_NvD3Component:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NvD3Component,
  data: {}
}
);
export function View_NvD3Component_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_NvD3Component_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'nvd3',([] as any[]),(null as any),(null as any),(null as any),View_NvD3Component_0,RenderType_NvD3Component)),
    import0.ɵdid(573440,(null as any),0,import1.NvD3Component,[import0.ElementRef],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const NvD3ComponentNgFactory:import0.ComponentFactory<import1.NvD3Component> = import0.ɵccf('nvd3',import1.NvD3Component,View_NvD3Component_Host_0,{
  options: 'options',
  data: 'data'
}
,{},([] as any[]));
